upload/extension/opencart/catalog/model/payment/opayo.php

```diff
@@ -0,0 +1,707 @@
+		$payment_data['RelatedVPSTxId'] = trim($opayo_order_info['VPSTxId'], '{}');
+		$payment_data['RelatedVendorTxCode'] = $opayo_order_info['VendorTxCode'];
+		$payment_data['RelatedSecurityKey'] = $opayo_order_info['SecurityKey'];
+		$payment_data['RelatedTxAuthNo'] = $opayo_order_info['TxAuthNo'];
+		$payment_data['COFUsage'] = 'SUBSEQUENT';
+		$payment_data['InitiatedType'] = 'MIT';
+		$payment_data['MITType'] = 'RECURRING';
+		$payment_data['RecurringExpiry'] = $recurring_expiry;
+		$payment_data['RecurringFrequency'] = $recurring_frequency;
+
+		if (!empty($order_info['shipping_lastname'])) {
+			$payment_data['DeliverySurname'] = substr($order_info['shipping_lastname'], 0, 20);
+			$payment_data['DeliveryFirstnames'] = substr($order_info['shipping_firstname'], 0, 20);
+			$payment_data['DeliveryAddress1'] = substr($order_info['shipping_address_1'], 0, 100);
+
+			if ($order_info['shipping_address_2']) {
+				$payment_data['DeliveryAddress2'] = $order_info['shipping_address_2'];
+			}
+
+			$payment_data['DeliveryCity'] = substr($order_info['shipping_city'], 0, 40);
+			$payment_data['DeliveryPostCode'] = substr($order_info['shipping_postcode'], 0, 10);
+			$payment_data['DeliveryCountry'] = $order_info['shipping_iso_code_2'];
+
+			if ($order_info['shipping_iso_code_2'] == 'US') {
+				$payment_data['DeliveryState'] = $order_info['shipping_zone_code'];
+			}
+
+			$payment_data['CustomerName'] = substr($order_info['firstname'] . ' ' . $order_info['lastname'], 0, 100);
+			$payment_data['DeliveryPhone'] = substr($order_info['telephone'], 0, 20);
+		} else {
+			$payment_data['DeliveryFirstnames'] = $order_info['payment_firstname'];
+			$payment_data['DeliverySurname'] = $order_info['payment_lastname'];
+			$payment_data['DeliveryAddress1'] = $order_info['payment_address_1'];
+
+			if ($order_info['payment_address_2']) {
+				$payment_data['DeliveryAddress2'] = $order_info['payment_address_2'];
+			}
+
+			$payment_data['DeliveryCity'] = $order_info['payment_city'];
+			$payment_data['DeliveryPostCode'] = $order_info['payment_postcode'];
+			$payment_data['DeliveryCountry'] = $order_info['payment_iso_code_2'];
+
+			if ($order_info['payment_iso_code_2'] == 'US') {
+				$payment_data['DeliveryState'] = $order_info['payment_zone_code'];
+			}
+
+			$payment_data['DeliveryPhone'] = $order_info['telephone'];
+		}
+
+		$response_data = $this->sendCurl($url, $payment_data, $i);
+
+		$response_data['VendorTxCode'] = $payment_data['VendorTxCode'];
+		$response_data['Amount'] = $payment_data['Amount'];
+		$response_data['Currency'] = $payment_data['Currency'];
+
+		return $response_data;
+	}
+
+	/**
+	 * Calculate Schedule
+	 *
+	 * @param string $frequency
+	 * @param string $next_payment
+	 * @param string $cycle
+	 *
+	 * @return
+	 */
+	private function calculateSchedule(string $frequency, string $next_payment, string $cycle) {
+		if ($frequency == 'semi_month') {
+			$day = date_format($next_payment, 'd');
```

Comment:
Use object oriented instead of procedural programming: https://www.php.net/manual/en/datetime.format.php
```suggestion
			$day = $next_payment->format('d');
```